### DISCUSSION TITLE
ESP32-S2's rmt(WS2812 Signal generation) seems to be messed up by TinyUSB

### DISCUSSION THREAD

**Discussion by 203Null** [hathach/tinyusb#1242] [General]
When I update LED while USB is communicating, the ws2812 signal seems to get messed up,
I was able to replicate this on both my own firmware with my own driver and TinyUF2.

Anyone has a clue what could be the cause and how can I fix it?

**Comment by 203Null** [comment]
I checked the signal with a logic analyzer, the timing seems fine but some packets are corrupted and there are extra 1s and 0s after where it suppose to finish.

**Comment by 203Null** [comment]
I was able to figure it out. The issue is due to RMT config. 
`config.mem_block_num = 3; ` (Default is 1)
fix the issue.

Not sure how this was related to the TinyUSB. Might just be a rtos/performance issue?

**Reply by hathach** [reply]
it is hard to say, you really need to use something like segger system view to see the rtos thread interaction to be sure. I am not sure of rmt implementation, but since usb irq and thread is typically have high priority it could potentially preempt and mess up the rmt timing. Try to have some critical section guard in your rmt timing sensitive to see if that helps.
